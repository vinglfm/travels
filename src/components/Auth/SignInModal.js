import React, {Component} from 'react';
import PropTypes from 'prop-types';
import Dialog from '@material-ui/core/Dialog';
import DialogTitle from '@material-ui/core/DialogTitle';
import DialogContent from '@material-ui/core/DialogContent';
import LogIn from './LogIn.js';
import SignIn from './SignIn.js';
import SignUp from './SignUp.js';
import styles from './SignInModal.css';


export default class SignInModal extends Component {
    constructor(props) {
        super(props);
        this.state = {
            component: 'SignIn'
        };

        this.getComponent = this.getComponent.bind(this);
        this.logInComponent = this.logInComponent.bind(this);
        this.signUpComponent = this.signUpComponent.bind(this);
        this.baseComponent = this.baseComponent.bind(this);
    }

    logInComponent(componentName) {
        this.setState({
            component: 'LogIn'
        });
    }

    signUpComponent(componentName) {
        this.setState({
            component: 'SignUp'
        });
    }

    baseComponent() {
        this.setState({
            component: 'SignIn'
        });
    }
  
    getComponent() {
        switch(this.state.component) {
            case 'LogIn':
                return <LogIn baseComponent={this.baseComponent}/>;
            case 'SignUp':
                return <SignUp baseComponent={this.baseComponent}/>;
            default:
                return <SignIn logInComponent={this.logInComponent} signUpComponent={this.signUpComponent} onSignIn={this.props.handleClose}/>;
        }
    }

    render() {
        return (
            <Dialog
            open={this.props.open}
            onEnter={this.baseComponent}
            onClose={this.props.handleClose}>
            <DialogTitle>
                <div className={styles.dialog__title}>Continue with</div>
            </DialogTitle>
            <DialogContent>
                {this.getComponent()}
            </DialogContent>
          </Dialog>
        );
    }
}

SignInModal.propTypes = {
    open: PropTypes.bool.isRequired,
    handleClose: PropTypes.func.isRequired
};