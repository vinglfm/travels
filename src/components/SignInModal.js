import React, {Component} from 'react';
import PropTypes from 'prop-types';
import Dialog from 'material-ui/Dialog';

  
export default class SignInModal extends Component {
    render() {
        return (
            <Dialog
            open={this.props.open}
            onRequestClose={this.props.handleClose}
            aria-labelledby='form-dialog-title'>
            Sign In dialog
          </Dialog>
        );
    }
}

SignInModal.propTypes = {
    open: PropTypes.bool.isRequired,
    handleClose: PropTypes.func.isRequired
};