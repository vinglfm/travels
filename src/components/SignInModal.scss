$focusColor: #88D5E9;
$submitColor: #3b5998;
$inputColor: #BEBEBE;

:local .dialog__title {
  text-align: center;
}

:local .fb__button {
    font-family: Helvetica, sans-serif;
    font-weight: 500;
    font-size: 1.2rem;
    background-color: #4c69ba;
    color: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    cursor: pointer;
    display: inline-block;
    text-decoration: none;
    text-transform: uppercase;
    background-color: #4c69ba;
    padding: .6rem;
}

:local .lines {
    margin: 1rem;
    line-height: 0.5;
    text-align: center;
    font-family: Helvetica, sans-serif;
    font-weight: 500;
    font-size: 1.2rem;

    span {
        display: inline-block;
        position: relative;  

        &:before,
        &:after {
            content: "";
            position: absolute;
            height: 0.1rem;
            border-bottom: 1px solid black;
            border-top: 1px solid black;
            top: 50%;
            width: 3rem;
        }

        &:before {
            right: 100%;
            margin-right: 2px;
        }

        &:after {
            left: 100%;
            margin-left: 2px;
        }
    }
  }

:local .log__in {
    font-style: italic;
    text-align: center;
    a {
        font-weight: bold;
        font-size: 1.2rem;
        cursor: pointer;
    }
}

:local .sign__up {
  font-style: italic;
  font-size: .5rem;

  a {
    font-weight: bold;
    font-size: .8rem;
    cursor: pointer;
  }
}

:local .modal {
    font: 1rem "Lucida Sans Unicode", "Lucida Grande", sans-serif;
  
    &__back {
      float: left;
      cursor: pointer;
    }
    
    &__header {
      text-align: center;
      font-size: 1rem;
      font-weight: bold;
    }
  
    &__divider {
      border-width: .5px;
      border-color: $inputColor;
    }
  
    &__form {
      margin: 10px auto;
      max-width: 400px;
  
      &__input {
        box-sizing: border-box;
        border: 1px solid $inputColor;
        padding: 7px;
        width: 97%;
        margin: 0;
        outline: none;
  
          &:focus {
            -moz-box-shadow: 0 0 8px $focusColor;
            -webkit-box-shadow: 0 0 8px $focusColor;
            box-shadow: 0 0 8px $focusColor;
            border: 1px solid $focusColor;
          }
      }
  
      &__label {
        font-size: .6rem;
        margin:0 0 3px 0;
        padding: 0;
        display: block;
        font-weight: bold;
  
        &__required {
          color: red;
        }
      }
  
      &__input__small {
        @extend .modal__form__input;
        width: 48%;
      }
  
      &__group {
        text-align: center;
        padding-top: 10px;
  
        &__submit {
          background: $submitColor;
          border: 1px solid rgba(0, 0, 0, 0.1);
          border-radius: .25rem;
          color: white;
          padding: 6px 12px;
          cursor: pointer;
          
          &:disabled {
            opacity: .6;
            cursor: not-allowed;
          }
        }
      }
    }
  }