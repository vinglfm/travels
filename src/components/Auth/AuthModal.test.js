import enzyme, {mount, shallow} from 'enzyme';
import ReactSixteenAdapter from 'enzyme-adapter-react-16';
enzyme.configure({ adapter: new ReactSixteenAdapter() });

import React from 'react';
import SignInModal from './AuthModal';

it('renders SignInModal in base state', () => {
  const snapshot = shallow(<SignInModal open handleClose={()=>{}} />);
  expect(snapshot).toMatchSnapshot();
});

it('renders SignInModal in logIn state', () => {
    const snapshot = shallow(<SignInModal open handleClose={()=>{}} />);
    snapshot.instance().logInComponent();
    snapshot.update();
    expect(snapshot).toMatchSnapshot();
  });

  it('renders SignInModal in signUp state', () => {
    const snapshot = shallow(<SignInModal open handleClose={()=>{}} />);
    snapshot.instance().signUpComponent();
    snapshot.update();
    expect(snapshot).toMatchSnapshot();
  });
