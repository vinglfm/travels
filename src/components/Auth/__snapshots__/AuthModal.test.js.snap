// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AuthModal 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AuthModal
    handleClose={[Function]}
    open={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(DialogTitle)>
          <div
                    className={undefined}
          >
                    Continue with
          </div>
</WithStyles(DialogTitle)>,
        <WithStyles(DialogContent)>
          <Connect(SignIn)
                    onSignIn={[Function]}
          />
</WithStyles(DialogContent)>,
      ],
      "fullWidth": true,
      "maxWidth": "xs",
      "onClose": [Function],
      "onEnter": undefined,
      "open": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            className={undefined}
>
            Continue with
</div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Continue with",
            "className": undefined,
          },
          "ref": null,
          "rendered": "Continue with",
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Connect(SignIn)
            onSignIn={[Function]}
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onSignIn": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(DialogTitle)>
            <div
                        className={undefined}
            >
                        Continue with
            </div>
</WithStyles(DialogTitle)>,
          <WithStyles(DialogContent)>
            <Connect(SignIn)
                        onSignIn={[Function]}
            />
</WithStyles(DialogContent)>,
        ],
        "fullWidth": true,
        "maxWidth": "xs",
        "onClose": [Function],
        "onEnter": undefined,
        "open": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <div
              className={undefined}
>
              Continue with
</div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Continue with",
              "className": undefined,
            },
            "ref": null,
            "rendered": "Continue with",
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(SignIn)
              onSignIn={[Function]}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onSignIn": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
